code/modules/fishing/fishing_rod.dm

```diff
@@ -0,0 +1,484 @@
+/obj/item/fishing_rod/proc/fish_bonus(fish_type)
+	return 0
+
+/obj/item/fishing_rod/proc/consume_bait()
+	if(bait)
+		QDEL_NULL(bait)
+		update_icon()
+
+/obj/item/fishing_rod/interact(mob/user)
+	if(currently_hooked_item)
+		reel(user)
+
+/obj/item/fishing_rod/proc/reel(mob/user)
+	//Could use sound here for feedback
+	if(do_after(user, 1 SECONDS, currently_hooked_item))
+		// Should probably respect and used force move later
+		step_towards(currently_hooked_item, get_turf(src))
+		if(get_dist(currently_hooked_item,get_turf(src)) < 1)
+			clear_hooked_item()
+
+/obj/item/fishing_rod/attack_self_secondary(mob/user, modifiers)
+	. = ..()
+	ui_interact(user)
+
+/obj/item/fishing_rod/pre_attack(atom/A, mob/living/user, params)
+	. = ..()
+	/// Reel in if able
+	if(currently_hooked_item)
+		reel(user)
+		return TRUE
+	SEND_SIGNAL(A, COMSIG_PRE_FISHING)
+
+/// Generates the fishing line visual from the current user to the target and updates inhands
+/obj/item/fishing_rod/proc/create_fishing_line(atom/movable/target, target_py = null)
+	var/mob/user = loc
+	if(!istype(user))
+		return
+	var/beam_color = line?.line_color || default_line_color
+	var/datum/beam/fishing_line/fishing_line_beam = new(user, target, icon_state = "fishing_line", beam_color = beam_color, override_target_pixel_y = target_py)
+	fishing_line_beam.lefthand = user.get_held_index_of_item(src) % 2 != 0
+	RegisterSignal(fishing_line_beam, COMSIG_BEAM_BEFORE_DRAW, .proc/check_los)
+	RegisterSignal(fishing_line_beam, COMSIG_PARENT_QDELETING, .proc/clear_line)
+	fishing_lines += fishing_line_beam
+	INVOKE_ASYNC(fishing_line_beam, /datum/beam/.proc/Start)
+	user.update_inv_hands()
+	return fishing_line_beam
+
+/obj/item/fishing_rod/proc/clear_line(datum/source)
+	SIGNAL_HANDLER
+	fishing_lines -= source
+	if(ismob(loc))
+		var/mob/user = loc
+		user.update_inv_hands()
+
+/obj/item/fishing_rod/dropped(mob/user, silent)
+	. = ..()
+	if(currently_hooked_item)
+		clear_hooked_item()
+	for(var/datum/beam/fishing_line in fishing_lines)
+		SEND_SIGNAL(fishing_line, COMSIG_FISHING_LINE_SNAPPED)
+	QDEL_LIST(fishing_lines)
+
+/// Hooks the item
+/obj/item/fishing_rod/proc/hook_item(mob/user, atom/movable/target)
+	if(currently_hooked_item)
+		return
+	if(!can_be_hooked(target))
+		return
+	currently_hooked_item = target
+	hooked_item_fishing_line = create_fishing_line(target)
+	RegisterSignal(hooked_item_fishing_line, COMSIG_FISHING_LINE_SNAPPED, .proc/clear_hooked_item)
+
+/// Checks what can be hooked
+/obj/item/fishing_rod/proc/can_be_hooked(atom/movable/target)
+	// Could be made dependent on actual hook, ie magnet to hook metallic items
+	return istype(target, /obj/item)
+
+/obj/item/fishing_rod/proc/clear_hooked_item()
+	SIGNAL_HANDLER
+
+	if(!QDELETED(hooked_item_fishing_line))
+		QDEL_NULL(hooked_item_fishing_line)
+	currently_hooked_item = null
+
+// Checks fishing line for interruptions and range
+/obj/item/fishing_rod/proc/check_los(datum/beam/source)
+	SIGNAL_HANDLER
+	. = NONE
```

Comment:
what is the point of this . = NONE?

Reply:
I believe it's good idea to make it explicit when the signal expects return value